app/models/concerns/domain/lockable.rb

```diff
@@ -0,0 +1,49 @@
+module Concerns::Domain::Lockable
```

Comment:
Given `ActiveRecord` has `lock!`, wouldn't it be better to name it `RegistryLockable` then?